torch/csrc/jit/codegen/cuda/python_frontend/fusion_manager.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <c10/macros/Export.h>
+
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h>
+
+#include <memory>
+
+//! nvFuser Fusion IR namespace abbreviation
+namespace Nvf = torch::jit::fuser::cuda;
+
+namespace nvfuser {
+
+struct RecordFunctor;
+
+//! \struct FusionCacheEntry
+//! \brief Is the container for a Node in the cache contained in the
+//! FusionManager that is organized as a prefix tree.
+
+struct TORCH_CUDA_CU_API FusionCacheEntry {
+  FusionCacheEntry(RecordFunctor* rec, bool _is_terminal = false);
+
+  //! An entry's primary data is the record it holds
+  std::unique_ptr<RecordFunctor> record;
+  //! A hash map of the children for the current node.
+  //! The hash map hashs a pointer to a RecordFunctor because
+  //! the hash function is virtual.
+  std::unordered_map<RecordFunctor*, std::unique_ptr<FusionCacheEntry>>
+      record_hash_map;
+
+  //! This boolean indicates a leaf node with a cached nvFuser Fusion
+  bool is_terminal;
```

Comment:
IIUC, this is always associated with
```C++
record->recordType() == RecordType::End
```
if this is the case, does it make sense to make this a method?
```C++
bool is_terminal() {
  return record->recordType() == RecordType::End;
}
```
This would simplify the rest part of the codebase a bit.

Reply:
I put in this simplification!  Thanks for the idea.